### PR TITLE
Manual Tally Functional Revamp

### LINKED ISSUE #3445: Manual Tallies: allow adding tally data for any voting method
## Summary

Currently, you can either add manual tally data as absentee ballots or precinct ballots, but not both at once. Will require some UI tweaks that dovetail with #3444.

### PR DESCRIPTION
**Pull request by adghayes**
## Overview

Closes #3445, #3444, and #3031. Updates how our manual tallies are split and adds a non-strict validation layer (does not prevent user saving invalid tallies). Changes the UI but did not focus on that, as @kofi-q will eventually make his way to these screens. 

@kofi-q where they existed, I tried to use new VVSG elements (mostly typography) but the two pages have table elements and select elements, which we don't have components for (yet), so there I just styled within the current framework.

## Demo Video or Screenshot

**Before Video:**

https://github.com/votingworks/vxsuite/assets/37960853/66afe113-7ab0-48a9-a419-e14a84ca88e4

**After Video Demo:**

https://github.com/votingworks/vxsuite/assets/37960853/d51d59f0-ab6d-481f-aaff-5934767d0bc4

**After Screenshot (summary page with no tallies added):**

Screen Shot 2023-05-18 at 7 15 14 PM

## Testing Plan
- Manual Testing
- Rewrite a bunch of automated testing

## Checklist

- [x] ~~I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.~~ 

- [x] I have added a screenshot and/or video to this PR to demo the change
- [x] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

File: apps/admin/frontend/src/components/ballot_counts_table.test.tsx [hunk 1]
```diff
@@ -54,15 +54,15 @@ describe('Ballot Counts by Precinct', () => {
 
   const manualResultsByPrecinct: Dictionary<ManualTally> = {
     // French Camp
-    '6526': fakeManualTally({
+    '6526': getMockManualTally({
```
Comment (adghayes) [inline apps/admin/frontend/src/components/ballot_counts_table.test.tsx:57]: Per #1139.

**Comment by adghayes** [comment]
@mattroe @carolinemodic @kofi-q could I get feedback on:
- places where I modified layout of table
- general UX
- how in-your-face validation is
- copy

File: apps/admin/frontend/src/screens/manual_data_entry_screen.tsx [hunk 2]
```diff
@@ -264,49 +264,58 @@ export function getContestTallyWithUpdatedNumberOfBallots(
...
   const { election } = electionDefinition;
-  const { precinctId: currentPrecinctId } =
-    useParams<ManualDataEntryScreenProps>();
+  const {
+    precinctId,
+    ballotStyleId,
+    ballotType: ballotTypeParam,
+  } = useParams<ManualDataEntryScreenProps>();
+  const precinct = find(election.precincts, (p) => p.id === precinctId);
+  const ballotStyle = find(
```
Comment (adghayes) [inline apps/admin/frontend/src/screens/manual_data_entry_screen.tsx:285]: We used to have a "could not find this precinct" page but I don't think that really makes sense. We should be, as this is doing implicitly, `assert`ing that the ids in the URL exist. I don't even know how to navigate to any specified URL in `kiosk-browser`.

**Comment by adghayes** [comment]
@mattroe sgtm, made the suggested changes:

Copy Change: 

Screen Shot 2023-05-21 at 12 55 25 PM


Delete Buttons:

Screen Shot 2023-05-21 at 12 57 04 PM

Delete Confirmation Modal:

Screen Shot 2023-05-21 at 12 57 16 PM

**Comment by kofi-q** [comment]
Design LGTM after the most recent updates 👍

**Comment by adghayes** [comment]
@kofi-q review by commit! I can reorganize things by file section but the chronological commit history is not bad, so I'll leave it as is for now.

**Review by kofi-q — APPROVED** [review]
Few minor comments, but LGTM!
Thanks for test-driving the new typography components - LMK if you have any dev experience feedback as you use them 🙏 

Also, probably not saying anything new, but general recommendation to try to break up changes into smaller PRs (tougher for big refactors, but worth the effort, I think)

File: apps/admin/frontend/src/screens/manual_data_entry_screen.tsx [hunk 3]
```diff
@@ -457,214 +471,257 @@ export function ManualDataEntryScreen(): JSX.Element {
...
+                      </Font>
+                    ) : (
+                      <Font color="warning">
+                        <Icons.Warning />
+                      </Font>
+                    )}
+                  </TD>
+                  <TD>
+                    {' '}
+                    <Font italic>
```
[the remark this answers is not in the document]
Comment (adghayes) [reply apps/admin/frontend/src/screens/manual_data_entry_screen.tsx]: Cool, updated in [19e6832](https://github.com/votingworks/vxsuite/pull/3478/commits/19e6832cd6c65d1bb16a5d9c35b3cebc12da381c).

Screen Shot 2023-05-23 at 2 06 28 PM

I agree the icon is enough below each contest. For the validation summary text on the bottom I made it semibold. Looks OK normal but I felt like it needs a little something more than the button color to call attention.

File: apps/admin/frontend/src/screens/manual_data_entry_screen.tsx [hunk 4]
```diff
@@ -457,214 +471,257 @@ export function ManualDataEntryScreen(): JSX.Element {
...
+                  </TD>
+                </tr>
+              </tfoot>
+            </Table>
+          </ContestData>
+        );
+      })}
+      <P>
+        {someContestHasInvalidResults ? (
+          <Font italic>
```
[the remark this answers is not in the document]
Comment (adghayes) [reply apps/admin/frontend/src/screens/manual_data_entry_screen.tsx]: [19e6832](https://github.com/votingworks/vxsuite/pull/3478/commits/19e6832cd6c65d1bb16a5d9c35b3cebc12da381c)

File: apps/admin/frontend/src/screens/manual_data_summary_screen.tsx [hunk 5]
```diff
@@ -1,134 +1,347 @@
...
+                        to={routerPaths.manualDataEntry({
+                          ballotStyleId: selectedBallotStyle.id,
+                          precinctId: selectedPrecinct.id,
+                          ballotType: selectedBallotType,
+                        })}
+                      >
+                        Add Results
+                      </LinkButton>
+                    ) : (
+                      <DecoyButton small disabled>
```
[the remark this answers is not in the document]
Comment (adghayes) [reply apps/admin/frontend/src/screens/manual_data_summary_screen.tsx]: Ah, sorry this commit is outdated. I have it as a `LinkButton` in a later commit when I found it made more sense.

**Pull request merged**